state-sync/state-sync-v2/state-sync-driver/src/driver.rs

```diff
@@ -35,78 +40,105 @@ pub struct DriverConfiguration {
 }
 
 impl DriverConfiguration {
-    pub fn new(role: RoleType, waypoint: Waypoint) -> Self {
-        Self { role, waypoint }
+    pub fn new(config: StateSyncDriverConfig, role: RoleType, waypoint: Waypoint) -> Self {
+        Self {
+            config,
+            role,
+            waypoint,
+        }
     }
 }
 
 /// The state sync driver that drives synchronization progress
-pub struct StateSyncDriver<M, S> {
-    // The component used to notify listeners of successful bootstrapping
-    bootstrap_notification_handler: BootstrapNotificationHandler,
+pub struct StateSyncDriver<D, M, S> {
```

Comment:
This is fine for now, but I do prefer named types that are more descriptive than a single letter, so I can tell that D means DataClient, M means Mempool Message?  S means StorageSynchronizer.

Reply:
Aah, fair point! I'll clean it up all in the next PR once I've added the bootstrapper (it has the same issue) 😄